vcap_utils/vcap_utils/backends/base_tensorrt.py

```diff
@@ -0,0 +1,278 @@
+        # Transfer input data to the GPU.
+        self.ctx.push()
+        [cuda.memcpy_htod_async(inp.device, inp.host, stream) for inp in inputs]
+        # Run inference.
+        # todo: use async or sync api?
+        # According to https://docs.nvidia.com/deeplearning/tensorrt/best-practices/index.html#optimize-python
+        # the performance should be almost identical
+        self.context.execute(
+            batch_size=batch_size, bindings=bindings
+        )
+        # Transfer predictions back from the GPU.
+        [cuda.memcpy_dtoh_async(out.host, out.device, stream) for out in outputs]
+        # Synchronize the stream
+        stream.synchronize()
+        # Return only the host outputs.
+        batch_outputs = []
+        for out in outputs:
+            entire_out_array = np.array(out.host)
+            out_array_by_batch = np.split(entire_out_array, batch_size)
+            out_lists = [out_array.tolist() for out_array in out_array_by_batch]
+            batch_outputs.append(out_lists)
+        final_outputs = []
+        for i in range(len(batch_outputs[0])):
+            final_output = []
+            for batch_output in batch_outputs:
+                final_output.append(batch_output[i])
+            final_outputs.append(final_output)
+        self.ctx.pop()
+        return final_outputs
+
+    def _prepare_post_process(self):
+        self.stride = 16
+        self.box_norm = 35.0
+        self.grid_h = int(self.engine_height / self.stride)
+        self.grid_w = int(self.engine_width / self.stride)
+        self.grid_size = self.grid_h * self.grid_w
+
+        self.grid_centers_w = []
+        self.grid_centers_h = []
+
+        for i in range(self.grid_h):
+            value = (i * self.stride + 0.5) / self.box_norm
+            self.grid_centers_h.append(value)
+
+        for i in range(self.grid_w):
+            value = (i * self.stride + 0.5) / self.box_norm
+            self.grid_centers_w.append(value)
+
+    def _apply_box_norm(self, o1: float, o2: float, o3: float, o4: float, x: int, y: int) -> \
+            Tuple[float, float, float, float]:
+        """
+        Applies the GridNet box normalization
+        Args:
+            o1 (float): first argument of the result
+            o2 (float): second argument of the result
+            o3 (float): third argument of the result
+            o4 (float): fourth argument of the result
+            x: row index on the grid
+            y: column index on the grid
+
+        Returns:
+            float: rescaled first argument
+            float: rescaled second argument
+            float: rescaled third argument
+            float: rescaled fourth argument
+        """
+        o1 = (o1 - self.grid_centers_w[x]) * -self.box_norm
+        o2 = (o2 - self.grid_centers_h[y]) * -self.box_norm
+        o3 = (o3 + self.grid_centers_w[x]) * self.box_norm
+        o4 = (o4 + self.grid_centers_h[y]) * self.box_norm
+        return o1, o2, o3, o4
```

Comment:
This section is hard to read. Also, I think the function should return int's, because the only time it's used we immediately convert the results to int (also having float values for pixels is probably a bad idea). I also updated the variable names for clarity.

Thoughts?

```suggestion
    def _apply_box_norm(self, o1: float, o2: float, o3: float, o4: float, x: int, y: int) -> \
            Tuple[int, int, int, int]:
        """
        Applies the GridNet box normalization
        Args:
            o1 (float): first argument of the result
            o2 (float): second argument of the result
            o3 (float): third argument of the result
            o4 (float): fourth argument of the result
            x: row index on the grid
            y: column index on the grid

        Returns:
            int: rescaled first argument
            int: rescaled second argument
            int: rescaled third argument
            int: rescaled fourth argument
        """
        xmin = int((o1 - self.grid_centers_w[x]) * -self.box_norm)
        ymin = int((o2 - self.grid_centers_h[y]) * -self.box_norm)
        xmax = int((o3 + self.grid_centers_w[x]) * self.box_norm)
        ymax = int((o4 + self.grid_centers_h[y]) * self.box_norm)
        return xmin, ymin, xmax, ymax
```

Reply:
Agree. I'll commit your change